### ISSUE TITLE
Insufficient accuracy of objective function in Elastic Net on some cases

### ISSUE THREAD

**Issue by agorshk** [oneapi-src/oneDAL#495]
Scikit-learn test "test_warm_start_convergence_with_regularizer_decrement" shows insufficient accuracy of objective function in Elastic Net algorithm on some problems (for example, in case of nonzero inputArgument of optimization solver algorithm). Differences of objective function between oneDAL and Sklearn implementation is more than tolerance. It looks like additional convergence control is needed (like in Sklearn, dual_gap).

![warm_start](https://user-images.githubusercontent.com/55500560/78027676-6c5b4280-7366-11ea-954d-153b2680d9e0.jpg)
